tests/lib/rules/jsx-curly-brace-presence.js

```diff
@@ -228,166 +234,218 @@ ruleTester.run('jsx-curly-brace-presence', rule, {
       errors: [
         {message: unnecessaryCurlyMessage}, {message: unnecessaryCurlyMessage}
       ],
       options: [{props: 'never'}]
     },
     {
-      code: '<App prop=\'foo\' attr=\"bar\" />',
-      output: '<App prop={\"foo\"} attr={\"bar\"} />',
+      code: `<App prop='foo' attr="bar" />`,
+      output: '<App prop={"foo"} attr={"bar"} />',
       errors: [
         {message: missingCurlyMessage}, {message: missingCurlyMessage}
       ],
       options: [{props: 'always'}]
     },
     {
-      code: '<App prop=\'foo\' attr={\"bar\"} />',
-      output: '<App prop={\"foo\"} attr={\"bar\"} />',
+      code: `<App prop='foo' attr={"bar"} />`,
+      output: `<App prop={"foo"} attr={"bar"} />`,
       errors: [{message: missingCurlyMessage}],
       options: [{props: 'always'}]
     },
     {
-      code: '<App prop={\'foo\'} attr=\'bar\' />',
-      output: '<App prop={\'foo\'} attr={\"bar\"} />',
+      code: `<App prop={'foo'} attr='bar' />`,
+      output: `<App prop={'foo'} attr={"bar"} />`,
       errors: [{message: missingCurlyMessage}],
       options: [{props: 'always'}]
+    },
+    {
+      code: `<App prop='foo &middot; bar' />`,
+      errors: [{message: missingCurlyMessage}],
+      options: [{props: 'always'}]
+    },
+    {
+      code: '<App>foo &middot; bar</App>',
+      errors: [{message: missingCurlyMessage}],
+      options: [{children: 'always'}]
+    },
+    {
+      code: `<App>{'foo "bar"'}</App>`,
+      output: `<App>foo "bar"</App>`,
+      errors: [{message: unnecessaryCurlyMessage}],
+      options: [{children: 'never'}]
+    },
+    {
+      code: `<App>{"foo 'bar'"}</App>`,
+      output: `<App>foo 'bar'</App>`,
+      errors: [{message: unnecessaryCurlyMessage}],
+      options: [{children: 'never'}]
+    },
+    {
+      code: [
+        '<App prop="    ',
+        '   a     ',
+        '     b      c',
+        '        d',
+        '">',
+        '  a',
+        '      b     c   ',
+        '         d      ',
+        '</App>'
+      ].join('\n'),
+      output: [
+        '<App prop={"    ',
+        '   a     ',
+        '     b      c',
+        '        d',
+        '"}>{"\\n  a\\n      b     c   \\n         d      \\n"}</App>'
+      ].join('\n'),
+      errors: [
+        {message: missingCurlyMessage}, {message: missingCurlyMessage}
+      ],
+      options: ['always']
+    },
+    {
+      code: [
+        `<App prop='    `,
+        '   a     ',
+        '     b      c',
+        '        d',
+        `'>`,
+        '  a',
+        '      b     c   ',
+        '         d      ',
+        '</App>'
+      ].join('\n'),
+      output: [
+        '<App prop={"    ',
+        '   a     ',
+        '     b      c',
+        '        d',
+        '"}>{"\\n  a\\n      b     c   \\n         d      \\n"}</App>'
+      ].join('\n'),
+      errors: [
+        {message: missingCurlyMessage}, {message: missingCurlyMessage}
+      ],
+      options: ['always']
```

Comment:
Check out the [babel repl](http://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&code_lz=DwQQBADgTg9hC8AiAUGMBDVawCNtgGMt80ATZRAPi03z2yJO3OAHoRrkBuZZUSWAgDeAAxrFc-RkzDkRAX0qjxdKRPxz5bDkA&debug=false&circleciRepo=&evaluate=false&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0). It strips some whitespace. We either have to replicate that, or bail out.

Reply:
Thanks for pointing it out! If we were to bail out, I would check for multiple whitespaces?

Reply:
I _think_ we should bail out if there's a line terminator in there. Multiple spaces on a single line should be fine.